version: "3.7"

services:
  person-service:
    container_name: ${_Values_root_name}-application
    image:  ${_Values_image_repository}${_Values_image_native}:${_Values_image_tag}
    restart: 'no'
    ports:
      - "${_Values_service_port}:${_Values_service_port}"
    command: ["-Xmx${_Values_resources_requests_memory}"]

#    we cannot active the healthcheck because base native images does not have curl
#    healthcheck:
#      test: "curl -f http://localhost:${_Values_service_port}/actuator/health || exit 1"
#      interval: 10s
#      timeout: 10s
#      retries: 5
    volumes:
      - tmp:/tmp
      - applogs:/usr/share/logs
    networks:
      - ${_Values_root_name}-net
    environment:
      #application
      - TZ=Europe/Berlin
      - server.port=${_Values_service_port}
      - tomcat.max-threads=20
      - security.authentication.enabled=false
      #forwarding sub url
      - server.forward-headers-strategy=FRAMEWORK
      #monitoring
      - management.endpoint.health.show-details=always
      - management.endpoints.web.exposure.include=health,prometheus,heapdump
      #zipkin / jaeger
      - spring.zipkin.service.name=person-service
      - spring.zipkin.baseUrl=http://host.docker.internal:9411
      - spring.sleuth.enabled=true
      - spring.zipkin.enabled=true
      #persistence
      - spring.datasource.url=jdbc:postgresql://${_Values_root_name}-postgres:5432/persondb
      - spring.datasource.username=person-service
      - spring.datasource.password=person-service
      - database.provisioning.goals=-import-demo-data
      #adapter
      - adapter.calleeservice.url=http://host.docker.internal:50900
      - adapter.calleeservice.user=YWRtaW4=
      - adapter.calleeservice.password=YWRtaW4=
      - adapter.timeout=10000
      #log levels
      - logging.level.root=WARN
      - logging.level.org.springframework=INFO
      - logging.level.org.goafabric=DEBUG
      - logging.level.org.flywaydb.core="INFO"
      - logging.level.org.hibernate.SQL="INFO"
      - logging.level.org.goafabric.personservice.persistence.audit="INFO"
      #log file settings
      - logging.file.max-size=10MB
      - logging.file.name=/usr/share/logs/application.log

    depends_on:
      - postgres

  postgres:
    container_name: ${_Values_root_name}-postgres
    image: postgres:14.1
    restart: 'no'
    ports:
      - 50810:5432
    environment:
      - TZ=Europe/Berlin
      - POSTGRES_USER=person-service
      - POSTGRES_PASSWORD=person-service
      - POSTGRES_DB=persondb
    command: postgres -c 'shared_buffers=128MB' -c 'max_connections=30' -c 'shared_preload_libraries=pg_stat_statements'
    volumes:
      - pgdata:/var/lib/postgresql/data
#      - ../config/postgres/initdb/00_pg_statements.sql:/docker-entrypoint-initdb.d/00_pg_statements.sql
    networks:
      - ${_Values_root_name}-net

  nginx:
    container_name: ${_Values_root_name}-nginx
    image: nginx:1.20.2
    restart: 'no'
    ports:
      - 8081:80
      - 443:443
    networks:
      - ${_Values_root_name}-net
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
      - ./server.pem:/etc/ssl/server.pem
      - ./server.key:/etc/ssl/server.key
    depends_on:
      - person-service

volumes:
    tmp:
    applogs:
    pgdata:

networks:
  person-service-net:
    driver: bridge